File: 1291_sequential_digits/sequential_digits.py
from typing import List


def sequential_digits(low: int, high: int) -> List[int]:
    """
    Time: O(1)
    Space: O(1)
    """
    nums = "123456789"
    i, j = len(str(low)), len(str(high))
    res = []
    for size in range(i, j + 1):
        for start in range(10 - size):
            num = int(nums[start: start + size])
            if high >= num >= low:
                res.append(num)
    return res

File: 1291_sequential_digits/sequential_digits.java
import java.util.List
import java.util.ArrayList;

class Solution {
    // Time: O(1)
    // Space: O(1)
    public List<Integer> sequentialDigits(int low, int high) {
        List<Integer> res = new ArrayList<>();
        String nums = "123456789";
        int minDigits = String.valueOf(low).length(), maxDigits = String.valueOf(high).length();
        for (int i = minDigits; i < maxDigits + 1; i++) {
            for (int j = 0; j < 10 - i; j++) {
                int tmp = Integer.parseInt(nums.substring(j, j + i));
                if (tmp >= low && tmp <= high) {
                    res.add(tmp);
                }
            }

        }
        return res;
    }
}
